### DISCUSSION TITLE
MikroORM and PostGIS usage

### DISCUSSION THREAD

**Discussion by bryan-vh** [mikro-orm/mikro-orm#1967] [Q&A] [answered]
Hello,

If possible, I'd like to know if there's an example of how to add a Geometry type to an Entity? I tried running the migration and it seemed that it couldn't discover the "undefined" type.

I followed the general concepts addressed for Custom Types and for PostGIS translation, but unsure of how to actually add it as a column using the EntityDecorator properly. 

Essentially, I want to be able to do this: 

```
ALTER TABLE your_table ADD COLUMN geom geometry(Point, 4326);
```

from the entity itself rather than with custom SQL code (if possible).

My current code for the custom type is as follows:

```
export class Point {
  constructor(readonly latitude: number, readonly longitude: number) {}
}

export class Geometry extends Type<Point | undefined, string | undefined> {
  convertToDatabaseValue(value: Point | undefined): string {
    if (!value) {
      throw ValidationError.invalidType(Geometry, value, 'js')
    }

    return `ST_SetSRID(ST_MakePoint(${value.longitude}, ${value.latitude}))`
  }
}
```

In the Entity itself I have the property like this: 

```
@Property({ type: Geography })
  point = new Point(this.latitude, this.longitude)
```

where latitude and longitude are simple number columns already in the table.

**Comment by B4nan** [comment, marked as answer]
You should also implement `getColumnType` method that will return the exact column type you want:

```ts
export class Geometry extends Type<Point | undefined, string | undefined> {

  // ...

  getColumnType(): string {
    return 'geometry(Point, 4326)'; // or whatever you want the type to be - it will be used as is
  }

}
```

> I tried running the migration and it seemed that it couldn't discover the "undefined" type.

That's another issue, hard to say what you are doing wrong without seeing any details, but you have something wrong in the discovery. Start with enabling debug mode, also check the example apps.

**Reply by bryan-vh** [reply]
That worked! Turns out I just forgot the getColumnType() function which is why it wasn't propagating properly. Thank you!

**Reply by B4nan** [reply]
In v5 the default is always set to string/varchar, so it should be more predictable.
